File: 170119_ActivityInClass/app/src/main/java/com/mobiledevices/miguel/inclassapp1/MainActivity.java
package com.mobiledevices.miguel.inclassapp1;

import android.app.Activity;
import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import static android.R.id.button2;
import static android.icu.lang.UCharacter.GraphemeClusterBreak.T;
import static android.widget.Toast.makeText;

public class MainActivity extends AppCompatActivity {

    private static final int REQUEST_CODE = 0;
    TextView textView;
    Button button, button2;
    EditText editText;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        textView = (TextView)findViewById(R.id.textView);
        button = (Button)findViewById(R.id.button);
        button2 = (Button)findViewById(R.id.button2);
        editText = (EditText)findViewById(R.id.editText);

        textView.setText("This is a TextView");
        button.setText("This is a Button");
        editText.setText("This is a Edit Text");

        //Context - A class that keeps info about the device
        //Activity extends context (So we can use it trough polymorphism)
        //getContext() - will return a Context object with the same data
        Toast.makeText(this, "Hey guys", Toast.LENGTH_LONG).show();

        button2.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View view){
                Toast.makeText(getApplicationContext(), editText.getText(), Toast.LENGTH_SHORT).show();
            }
        });

    }

    //Catch button press
    public void buttonWasClicked(View v){
        Toast.makeText(this, "Button clicked", Toast.LENGTH_SHORT).show();

        //In Android to open up a new activity you don't COMMAND
        //You ASK!

        //Intent - Request for activity
        Intent intent = new Intent(this, MainActivity2.class);
        intent.putExtra("test", editText.getText().toString());
        startActivityForResult(intent, REQUEST_CODE);
    }

    protected void onActivityResult(int requestCode, int resultCode, Intent data){
        if(requestCode == REQUEST_CODE && resultCode == Activity.RESULT_OK){
            Toast.makeText(this, data.getStringExtra("result"), Toast.LENGTH_SHORT).show();
        }
    }
}

File: 170424_ActivityInClass/MainActivity.cs
using Android.App;
using Android.Widget;
using Android.OS;

namespace XamarinClaseMartes
{
    [Activity(Label = "XamarinClaseMartes", MainLauncher = true, Icon = "@drawable/icon")]
    public class MainActivity : Activity
    {
        private EditText ed1;
        private Button b1;
        private Button b2;

        protected override void OnCreate(Bundle bundle)
        {
            base.OnCreate(bundle);

            // Set our view from the "main" layout resource
            SetContentView (Resource.Layout.Main);

            ed1 = FindViewById<EditText>(Resource.Id.ed1);
            b1 = FindViewById<Button>(Resource.Id.b1);
            b2 = FindViewById<Button>(Resource.Id.b2);

            b1.Click += delegate
            {
                Toast.MakeText(this, ed1.Text, ToastLength.Short).Show();
            };

            b2.Click += delegate {
                JSONRequest request = new JSONRequest();
                request.Execute("http://jsonplaceholder.typicode.com/posts");

            };
        }
    }
}

